### ISSUE TITLE
Get Access to the Latest Internet Knowledge

### ISSUE THREAD

**Issue by franva** [imartinez/privateGPT#1239]
Wow great work~!!!
I like the idea of private GPT~! 

BUT there is one question need to be asked: How do I make sure the PrivateGPT has the most UP-TO-DATE Internet knowledge? like ChatGPT 4-Turob has knowledge up to April 2023. 
Surly we can use other open source LLM models and plugin networks to enable the LLM to search in Internet(I do not know how to do this, appreciate if anyone knows and could share the code) but they do not have the latest knowledge from the Internet.
This is important to many users as have access to the latest almost everything makes life easier.

Looking forward to a way(hopefully, it could be easy enough for individual users) to be able to update the knowledge base to the latest Internet.

**Comment by jackfood** [comment]
@franva PrivateGPT i believe is a storage of vector for you to retrieve. One way is to webcrawler website that you wish the information to have, store as vector, and query from there. There are some github projects (i think) that webcrawler data from internet. But data from internet can be subjects, do take note of the content and decide for yourself.

**Comment by franva** [comment]
Hi @jackfood  I do have this approach implemented. 
I use LangChain + Chromadb which stores the vectors I extracted from PDF files.

I am curious about the performance and accuracy between:
A: LLM which has the specific knowledge (e.g. a group of PDF files) already built-in
B: an LLM which has no specific knowledge about the PDF files, but it plugs in a Vector Store which stores the embeddings of those PDF files.

So I am curious about the performance and accuracy between A and B.

Any answers are welcomed.

**Comment by jackfood** [comment]
@franva Yes... The LLM possesses specific knowledge embedded in it when it is trained, but it might be outdated. Adding additional PDFs won't update the LLM's original knowledge. Instead, it extracts chunks of data from PDFs or texts, presenting the closest relevant information to your query. It then asks the LLM to summarize these chunks. Using one chunk may be quicker but lacks content. In contrast, using 10 chunks provides more extensive content and is closer to detail but demands more tokens and processing power. Considering our ctx limit of around 8000 tokens, using multiple chunks of about 4000-5000 tokens may yield a concise 3000-token answer, though it requires substantial processing power and depends on the model's training basis.
